src/main/kotlin/org/koitharu/kotatsu/parsers/site/HoneyMangaParser.kt

```diff
@@ -0,0 +1,210 @@
+	private val genresListApi = "$urlApi/genres-tags/genres-list"
+	private val framesApi = "$urlApi/chapter/frames"
+	private val searchApi = "https://search.api.honey-manga.com.ua/api/v1/title/search-matching?query="
+
+	private val imageStorageUrl = "https://manga-storage.fra1.digitaloceanspaces.com/public-resources"
+	override val headers
+		get() = Headers.Builder()
+			.add("User-Agent", "Mozilla/5.0 (Android 13; Mobile; rv:68.0) Gecko/68.0 Firefox/109.0")
+			.build()
+
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("honey-manga.com.ua", null)
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.POPULARITY,
+		SortOrder.NEWEST,
+	)
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.US)
+		val body = JSONObject()
+		body.put("mangaId", manga.url)
+		body.put("pageSize", 999999) // Hack lol (no)
+		body.put("page", 1)
+		body.put("sortOrder", "ASC")
+		val chapterRequest = webClient.httpPost(chapterApi, body).parseJson()
+		return manga.copy(
+			chapters = chapterRequest.getJSONArray("data").mapJSONIndexed() { i, jo ->
+				MangaChapter(
+					id = generateUid(jo.getString("id")),
+					name = buildString {
+						append("Том ")
+						append(jo.optString("volume", "0"))
+						append(". ")
+						append("Розділ ")
+						append(jo.optString("chapterNum", "0"))
+						if (jo.optString("title") != "Title") {
+							append(" - ")
+							append(jo.optString("title"))
+						}
+					},
+					number = i + 1,
+					url = jo.optString("chapterResourcesId"),
+					scanlator = null,
+					uploadDate = dateFormat.tryParse(jo.getString("lastUpdated")),
+					branch = null,
+					source = source
+				)
+			}
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder
+	): List<Manga> {
+		val body = JSONObject()
+		var content: JSONArray? = null
+		content = if (!query.isNullOrEmpty()) {
+			if((query.length < 3) || (page > 1)) return emptyList()
+			body.put("query", query)
+			webClient.httpGet(searchApi + query).parseJsonArray()
+		} else {
+			body.put("page", page)
+			body.put("pageSize", PAGE_SIZE)
+			body.put("filters", JSONArray())
+			val sort = JSONObject()
+			sort.put("sortBy", getSortKey(sortOrder))
+			sort.put("sortOrder", "DESC")
+			body.put("sort", sort)
+			webClient.httpPost(mangaApi, body).parseJson().getJSONArray("data")
+		}
+		return content!!.mapJSON { jo ->
+			val id = jo.getString("id")
+			val posterUrl = jo.getString("posterUrl")
+			Manga(
+				id = generateUid(id),
+				title = jo.getString("title"),
+				altTitle = jo.getStringOrNull("alternativeTitle"),
+				url = id,
+				publicUrl = "https://$domain/book/$id",
+				rating = RATING_UNKNOWN,
+				isNsfw = isNsfw(jo.getStringOrNull("adult")),
+				coverUrl = getCoverUrl(posterUrl, 256),
+				tags = getTags(jo.optJSONArray("genresAndTags")!!),
```

Comment:
```suggestion
				tags = jo.optJSONArray("genresAndTags")?.let(::getTags).orEmpty(),
```

Reply:
Why not keep it as a function? Renamed it to be more suitable for the function